packages/sanity/src/core/studio/studioAnnouncements/StudioAnnouncementsProvider.tsx

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable camelcase */
+import {useTelemetry} from '@sanity/telemetry/react'
+import {useCallback, useEffect, useMemo, useState} from 'react'
+import {StudioAnnouncementContext} from 'sanity/_singletons'
+
+import {useClient} from '../../hooks/useClient'
+import {useWorkspace} from '../../studio/workspace'
+import {SANITY_VERSION} from '../../version'
+import {
+  ProductAnnouncementCardClicked,
+  ProductAnnouncementCardDismissed,
+  ProductAnnouncementCardSeen,
+  ProductAnnouncementModalDismissed,
+} from './__telemetry__/studioAnnouncements.telemetry'
+import {StudioAnnouncementsCard} from './StudioAnnouncementsCard'
+import {StudioAnnouncementsDialog} from './StudioAnnouncementsDialog'
+import {
+  type DialogMode,
+  type StudioAnnouncementDocument,
+  type StudioAnnouncementsContextValue,
+} from './types'
+import {useSeenAnnouncements} from './useSeenAnnouncements'
+import {isValidAnnouncementAudience, isValidAnnouncementRole} from './utils'
+
+interface StudioAnnouncementsProviderProps {
+  children: React.ReactNode
+}
+const CLIENT_OPTIONS = {apiVersion: 'v2024-09-19'}
+
+/**
+ * @internal
+ * @hidden
+ */
+export function StudioAnnouncementsProvider({children}: StudioAnnouncementsProviderProps) {
+  const telemetry = useTelemetry()
+  const [dialogMode, setDialogMode] = useState<DialogMode | null>(null)
+  const [isCardDismissed, setIsCardDismissed] = useState(false)
+  const [studioAnnouncements, setStudioAnnouncements] = useState<StudioAnnouncementDocument[]>([])
+  const [seenAnnouncements, setSeenAnnouncements] = useSeenAnnouncements()
+  const {currentUser} = useWorkspace()
+  const client = useClient(CLIENT_OPTIONS)
+
+  const unseenAnnouncements: StudioAnnouncementDocument[] = useMemo(() => {
+    // If it's loading or it has errored return an empty array to avoid showing the card
+    if (seenAnnouncements.loading || seenAnnouncements.error) return []
+    // If none is seen, return all the announcements
+    if (!seenAnnouncements.value) return studioAnnouncements
+
+    // Filter out the seen announcements
+    const unseen = studioAnnouncements.filter((doc) => !seenAnnouncements.value?.includes(doc._id))
+    if (unseen.length > 0) {
+      telemetry.log(ProductAnnouncementCardSeen, {
+        announcement_id: unseen[0]._id,
+        announcement_title: unseen[0].title,
+        source: 'studio',
+        studio_version: SANITY_VERSION,
+      })
+    }
+    return unseen
+  }, [seenAnnouncements, studioAnnouncements, telemetry])
+
+  useEffect(() => {
```

Comment:
Just a thought—but not worth changing—you may be able to simplify this a little by using the `useObservable` hook from ReactRx and modelling it as a single observable that filters the response.

Usage could then look a bit like this:

```ts
const studioAnnouncements = useObservable(fetchAnnouncements)
```

Reply:
that's a very good suggestion and it simplified the mocking.  https://github.com/sanity-io/sanity/pull/7515/commits/ae0aa6a22dc7bdabaedeae0cf538e72ce7e4d7e5

Reply:
Nice! 🙌